### ISSUE TITLE
conv with 6 filters

### ISSUE THREAD

**Issue by squeakus** [matpalm/movidius_bug_reports#1]
Probably the opposite of an issue but I had a quick look at this one and it seems to be the default padding messing everything up again. I changed the conv_with_n to valid and it seems to fix it:

model = slim.conv2d(imgs, num_outputs=n, kernel_size=3, padding="VALID", stride=2, scope='e1')

**Comment by matpalm** [comment]
ok, that's great! ( and sorry, i did think i'd tried that combo in the past).

winding back a bit then the entire reason i had this test was that i was seeing problems when using `num_outputs=1`, that's actually the case i care about. 

so i'll update these tests this afternoon to more directly reflect these combos ..

* `conv_with_8_filters_and_padding_valid`
* `conv_with_8_filters_and_padding_same`
* `conv_with_1_filter_and_padding_valid`
* `conv_with_1_filter_and_padding_same`

thanks!

**Comment by matpalm** [comment]
added in c268a2343c0fcb306fa8477a5ea902f4b0776b42  i'll rework `deconv_padding_same` now with `num_filters=1` and `padding=VALID` ; if i can get that working it'll be the first minimal example of a structure the same as [BNN](https://github.com/matpalm/bnn)

**Comment by matpalm** [comment]
added  162bb86 which updates the `conv_deconv_output_shape_wrong` example with...

* padding=VALID everywhere
* a single kernel final conv layer for the 1x1 logit calculation

this is very close to the main model i want to get working now!! just one problem; the shape of the output from the NCS call is wrong. the final output should be `(31,31,1)` but it's `(29791,)` ...

i'll keep hacking for a workaround, but very close!

**Comment by matpalm** [comment]
ok! so i have a workaround for the `conv_deconv_output_shape_wrong` error...

though i want `(31,31,1)` if i flatten everything to `(961,)` it all works. 

so i'm guessing the NCS api is happy with a 1d but not 2d output? 

i'll add another test for that case, that demonstrates how it works, but will keep the failing one...

**Comment by matpalm** [comment]
so it looks like the bug is that the output shape is being set wrong...

the output from the NCS call is shaped `(29791,)` but if i take the first `31*31` entries and reshape myself it works ... 

`ncs_positive_prediction = ncs_positive_prediction[:31*31].reshape((31,31,1))`

... so the data is _there_, it's just the tensor shape/size isn't being set properly by the api

**Comment by matpalm** [comment]
added 28eab9e8695962ca9949b1ea989a339dcaa2697b  which includes the slice/reshape hack to get `conv_deconv_output_shape_wrong` working.

nice! i have everything i need to start porting a model of BNN to the compute stick;  but all done for today at least :)

**Comment by squeakus** [comment]
Okay this has really got my colleague's interest! The NCS has to do a lot of reshaping when processing the data and he thinks that when parsing and reshaping the 2D tensorflow network it is being applied on the wrong axis as 29791 == 31^3.....

initial investigation suggests it may have something to do with the last sigmoid layer. He recommends instead working with the logits layer directly after training. If you invert the sigmoid you can calculate the threshold and then work with the logits layer directly which is more efficient. I hope this helps!

**Comment by matpalm** [comment]
switching the output node to the logits (i.e. an output node of `logits/BiasAdd` for this model) doesn't make a difference to the shape; it still ends up being something i need to reshape.

however! interestingly it does help me in another way...

if i try to get an output larger than `(63,63,1)`, specifically `(127,127,1)`, i get a read error; see [my comments at the bottom of this forum post](https://ncsforum.movidius.com/discussion/928/dispatchereventreceive-read-failed-4)

however if i work on the logits i don't have the read error at `(127,127,1)` (even though i still do have the shape error)

so, i'm going to switch to not using the sigmoid, i can do that in postprocessing, but i'm doing it to avoid the read error, not the shape error :D
